src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts

```diff
@@ -1124,6 +1124,38 @@ describe("hubConnection", () => {
                 fail(e);
             }
         });
+
+        it("overwrites library headers with user headers", async (done) => {
+            const headers = { "User-Agent": "Custom Agent", "X-HEADER": "VALUE" };
+            const hubConnection = getConnectionBuilder(t, TESTHUBENDPOINT_URL, { headers })
+                .withHubProtocol(new JsonHubProtocol())
+                .build();
+
+            try {
+                await hubConnection.start();
+
+                const [name, value] = getUserAgentHeader();
```

Comment:
Call this before setting `headers` so you can use `name` to set the user agent header
```typescript
const [name, value] = getUserAgentHeader();
const headers = { name: "Custom Agent", "X-HEADER": "VALUE" };
```

That way you can simplify the logic below to be just `if ((t === HttpTransportType.ServerSentEvents || t === HttpTransportType.WebSockets) && !Platform.isNode)`